hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java

```diff
@@ -975,29 +978,51 @@ public CompletableFuture<Void> flush(TableName tableName, List<byte[]> columnFam
     // If the server version is lower than the client version, it's possible that the
     // flushTable method is not present in the server side, if so, we need to fall back
     // to the old implementation.
+    Preconditions.checkNotNull(columnFamilyList,
+      "columnFamily is null, If you don't specify a columnFamily, use flush(TableName) instead.");
     List<byte[]> columnFamilies = columnFamilyList.stream()
       .filter(cf -> cf != null && cf.length > 0).distinct().collect(Collectors.toList());
-    FlushTableRequest request = RequestConverter.buildFlushTableRequest(tableName, columnFamilies,
-      ng.getNonceGroup(), ng.newNonce());
-    CompletableFuture<Void> procFuture = this.<FlushTableRequest, FlushTableResponse> procedureCall(
-      tableName, request, (s, c, req, done) -> s.flushTable(c, req, done),
-      (resp) -> resp.getProcId(), new FlushTableProcedureBiConsumer(tableName));
     CompletableFuture<Void> future = new CompletableFuture<>();
-    addListener(procFuture, (ret, error) -> {
+    addListener(getDescriptor(tableName), (tDesc, error) -> {
```

Comment:
Introducing `getDescriptors` introduces an additional RPC. I think we shouldn't do this, only to do validation on client-side. Instead please post the original message and let the server validate it. If we had some client-side cache of the descriptor, we could do a pre-validation on that, but even then, caches are lossy and probably shouldn't be relied upon for something that the server is already the authority.

Reply:
I will refactor this part of the code later based on your good suggestions
Thank you both for your review